src/upload.rs

```diff
@@ -196,12 +199,23 @@ fn resolve_pypi_cred(
     opt: &PublishOpt,
     config: &Ini,
     registry_name: Option<&str>,
+    registry_url: &str,
 ) -> (String, String) {
     // API token from environment variable takes priority
     if let Ok(token) = env::var("MATURIN_PYPI_TOKEN") {
         return ("__token__".to_string(), token);
     }
 
+    // Try to get a token via OIDC exchange
+    match resolve_pypi_token_via_oidc(registry_url) {
```

Comment:
what happens if someone uses this with a registry that isn't pypi?

Reply:
It will print a failure warning, but we can certainly do better to not show a warning. I'll improve it later.

Reply:
https://github.com/PyO3/maturin/commit/6daaed5aac9f08820bd7cc2b5f3033ad470b5c30

Reply:
thank you!